src/components/Snippet/index.tsx

```diff
@@ -0,0 +1,219 @@
+  align-items: center;
+`
+
+const AnimatedArrowIcon = styled(Icon, {
+  shouldForwardProp: prop => !['showMore'].includes(prop),
+})<{ showMore?: boolean }>`
+  transform: ${({ showMore }) =>
+    showMore ? 'rotate(180deg)' : 'rotate(0deg)'};
+  transform-origin: center;
+  transition: transform 300ms ease-in-out;
+`
+
+type SnippetProps = {
+  className?: string
+  children: string
+  /**
+   * prefix display an element at the beginning of the snippet that is not copiable or selectable.
+   * For `lines` prefix it will display the line number.
+   * For `command` prefix it will display a `$` sign.
+   */
+  prefix?: Prefixes
+  showText?: string
+  hideText?: string
+} & Pick<ComponentProps<typeof CopyButton>, 'copyText' | 'copiedText'>
+
+const Snippet = ({
+  children,
+  copyText,
+  copiedText,
+  showText = 'Show',
+  hideText = 'Hide',
+  prefix,
+  className,
+}: SnippetProps) => {
+  const theme = useTheme()
+  const [showMore, setShowMore] = useState(false)
+
+  const lines = children.split(LINES_BREAK_REGEX).filter(Boolean)
+  const numberOfLines = lines.length
+  const multiline = numberOfLines > 1
+  const hasShowMoreButton = numberOfLines > 4 && multiline
+
+  /**
+   * This is used to calculate the max height of the snippet in order to add an animation when the show more button is clicked.
+   * The max height is calculated by multiplying the number of lines by the line height of the text plus the padding (2 times).
+   */
+  const maxHeightExtended =
+    numberOfLines * Number(theme.typography.code.lineHeight.replace('px', '')) +
+    Number(theme.space['2'].replace('px', '')) * 2
+
+  return (
+    <Container multiline={multiline} className={className}>
+      <StyledStack>
+        <PreText
+          as="pre"
+          variant="code"
+          multiline={multiline}
+          hasShowMoreButton={hasShowMoreButton}
+          showMore={showMore}
+          maxHeightExtended={maxHeightExtended}
+        >
+          {multiline ? (
+            Children.map(lines, child => (
+              <StyledSpan multiline prefix={prefix}>
+                {child}
+              </StyledSpan>
+            ))
+          ) : (
+            <StyledSpan prefix={prefix}>{children}</StyledSpan>
+          )}
+        </PreText>
+        <ButtonContainer multiline={multiline && numberOfLines > 1}>
+          <CopyButton
+            value={children}
+            copyText={copyText}
+            copiedText={copiedText}
+            noBorder
+            variant="neutral"
+          />
+        </ButtonContainer>
+        {hasShowMoreButton ? (
+          <ShowMoreContainer showMore={showMore}>
+            <StyledButton type="button" onClick={() => setShowMore(!showMore)}>
+              <AlignCenterText as="span" variant="bodySmallStrong">
+                {showMore ? hideText : showText}
+                &nbsp;
+                <AnimatedArrowIcon name="arrow-down" showMore={showMore} />
+              </AlignCenterText>
```

Comment:
(pseudo) 
```jsx
              <Stack>
                <Text>{showMore ? hideText : showText}</Text>
                &nbsp;
                <AnimatedArrowIcon name="arrow-down" showMore={showMore} />
              </Stack>
```

better syntax, less components

Reply:
Your solution is less semantically correct than the one @matthprost proposed I think

Reply:
Beg to differ, not much of semantic difference

```
<span>
  {text}
  <svg />
</span>
```
vs
```
<div>
  <p>{text}</p>
  <svg />
</div>
```

But it'll avoid a styled for nothing, and use our positioning components for what they were meant to do

Reply:
@chambo-e Those are into a Button, Button expect to have phrasing content and cannot have `div` as children
cf https://www.w3.org/TR/2012/WD-html5-20120329/the-button-element.html#the-button-element

That's why I made a styled 😄